Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProject/Screens/CourseDetailsVC.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  CourseDetailsVC.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 10/1/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import UIKit
+
+class CourseDetailsVC: UIViewController {
+  var courseContentUrl: String?
+  @IBOutlet var courseLogoImageView: UIImageView!
+  @IBOutlet var courseTitle: UILabel!
+  @IBOutlet var activityIndicator: UIActivityIndicatorView!
+  @IBOutlet var courseBody: UITextView!
+  @IBOutlet var courseDescription: UILabel!
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    loadCourseContent()
+  }
+
+  // MARK: - Load Course Content
+  private func loadCourseContent() {
```

Comment:
You can make this function a lot cleaner that it is right now. Here are a few suggestions:

1. Consider getting familiar with URLSession with Combine. It will make your dispatch group code obsolete, and won't require the temporary variables at the top of the function.
2. Consider moving the handlers for `fetchCourseContent` and `notify` into separate methods to make this function a bit easier to digest.
3. I'm not sure you need the dispatch group here? It seems you are already nesting requests, initiating one after the other, so the notify doesn't really do anything that you couldn't do in `fetchArtwork`'s callback.